File: src/TypeFormatter/AnyTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import { AnyType } from "../Type/AnyType.js";
import type { BaseType } from "../Type/BaseType.js";

export class AnyTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof AnyType;
    }
    public getDefinition(type: AnyType): Definition {
        return {};
    }
    public getChildren(type: AnyType): BaseType[] {
        return [];
    }
}

File: internal/formatter/any_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// AnyTypeFormatter mirrors src/TypeFormatter/AnyTypeFormatter.ts.
type AnyTypeFormatter struct{}

func NewAnyTypeFormatter() *AnyTypeFormatter { return &AnyTypeFormatter{} }

func (f *AnyTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.AnyType)
	return ok
}

func (f *AnyTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{}
}

func (f *AnyTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
